### PR TITLE
Stop using pip for 3.10 on public ci (Revert "start testing Python 3.10 on public CI")

### PR DESCRIPTION
**Pull request by esc**
This reverts commit 74b5a934206d9b01466336e77373b4a59d09a7e5.

Stop using `pip` to install dependencies on Python 3.10 on public CI.
This configuration was put in place during a time where not many conda
packages were available for Numba's dependencies. Ideally this should no
longer be the case. This PR will test that hypothesis and fix the
configuration once all packages do become available.

### CODE REVIEW

**Comment by esc** [comment]
There appears to be a reproducible error:

```
======================================================================
FAIL: test_fanout_2 (numba.tests.test_refop_pruning.TestRefOpPruning)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vsts/work/1/s/numba/tests/test_refop_pruning.py", line 123, in test_fanout_2
    fanout=False, fanout_raise=True)
  File "/home/vsts/work/1/s/numba/tests/test_refop_pruning.py", line 49, in check
    self.assertEqual(stat, 0, msg=msg)
AssertionError: 3 != 0 : failed checking fanout
```

But only on the `Linux py37_np120_svml` build.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -117,11 +117,6 @@ jobs:
         NUMPY: '1.21'
         CONDA_ENV: azure_ci
         TEST_START_INDEX: 16
-      py310_np121:
-        PYTHON: '3.10'
-        NUMPY: '1.21'
-        CONDA_ENV: azure_ci
-        TEST_START_INDEX: 17
```
[the remark this answers is not in the document]
Comment (esc) [reply azure-pipelines.yml:124]: no

File: buildscripts/incremental/setup_conda_environment.sh [hunk 2]
```diff
@@ -56,15 +52,13 @@ source activate $CONDA_ENV
...
 echo "VANILLA_INSTALL=$VANILLA_INSTALL"
-if [[ "$PYTHON" != "3.10" && "$VANILLA_INSTALL" != "yes" ]]; then
-    # Scipy, CFFI, jinja2, IPython, and pygments are optional dependencies,
-    # but exercised in the test suite.
+if [ "${VANILLA_INSTALL}" != "yes" ]; then
+    # Scipy, CFFI, jinja2, IPython, ipython_genutils and pygments are optional dependencies, but exercised in the test suite
     # pexpect is used to run the gdb tests.
     # ipykernel is used for testing ipython behaviours.
-    $CONDA_INSTALL ${EXTRA_CHANNELS} cffi jinja2 ipython ipykernel pygments pexpect
+    $CONDA_INSTALL ${EXTRA_CHANNELS} cffi jinja2 ipython ipython ipykernel pygments pexpect
```
Comment (stuartarchibald) [inline buildscripts/incremental/setup_conda_environment.sh:61]: This is adding `ipython` twice.

File: buildscripts/incremental/setup_conda_environment.sh [hunk 3]
```diff
@@ -56,15 +52,13 @@ source activate $CONDA_ENV
...
 
 # Install optional packages into activated env
-
 echo "PYTHON=$PYTHON"
 echo "VANILLA_INSTALL=$VANILLA_INSTALL"
-if [[ "$PYTHON" != "3.10" && "$VANILLA_INSTALL" != "yes" ]]; then
-    # Scipy, CFFI, jinja2, IPython, and pygments are optional dependencies,
-    # but exercised in the test suite.
+if [ "${VANILLA_INSTALL}" != "yes" ]; then
+    # Scipy, CFFI, jinja2, IPython, ipython_genutils and pygments are optional dependencies, but exercised in the test suite
```
Comment (stuartarchibald) [inline buildscripts/incremental/setup_conda_environment.sh:58]: line wrap?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, there's a couple of minor things to look at else looks good.

**Comment by esc** [comment]
@stuartarchibald I have addressed the comments and it looks like it is passing public ci now.

File: buildscripts/incremental/setup_conda_environment.sh [hunk 4]
```diff
@@ -56,12 +52,11 @@ source activate $CONDA_ENV
...
 
 # Install optional packages into activated env
-
 echo "PYTHON=$PYTHON"
 echo "VANILLA_INSTALL=$VANILLA_INSTALL"
-if [[ "$PYTHON" != "3.10" && "$VANILLA_INSTALL" != "yes" ]]; then
-    # Scipy, CFFI, jinja2, IPython, and pygments are optional dependencies,
-    # but exercised in the test suite.
+if [ "${VANILLA_INSTALL}" != "yes" ]; then
+    # Scipy, CFFI, jinja2, IPython, ipython_genutils and pygments are optional
```
Comment (esc) [inline buildscripts/incremental/setup_conda_environment.sh:58]: I wonder if I should add scipy back in here, it doesn't seem to be installed at all at present. Also, perhaps I should remove `ipython_genutils`?

Comment (stuartarchibald) [reply buildscripts/incremental/setup_conda_environment.sh:58]: Looks like SciPy is conditional on "not 32 bit linux"?
https://github.com/numba/numba/blob/73e84b50bd8a80980f0349528002fee7d653b014/buildscripts/incremental/setup_conda_environment.sh#L68-L72 ?

Comment (stuartarchibald) [reply buildscripts/incremental/setup_conda_environment.sh:58]: RE: `ipython_genutils`. I think what is in `main` is correct:
https://github.com/numba/numba/blob/73e84b50bd8a80980f0349528002fee7d653b014/buildscripts/incremental/setup_conda_environment.sh#L63-L66

Comment (esc) [reply buildscripts/incremental/setup_conda_environment.sh:58]: Yes, it makes sense. I will remove `ipython_genutils` from the comment.

**Pull request merged**